### PR TITLE
Fix issue causing node with default config fail on startup 

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Fixes [11572](https://github.com/crate/crate/issues/11572)

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Review by mfussenegger — COMMENTED** [review]
This would also invalidate the docs and leave the node-to-node SSL encryption without any authentication support.

We'll have to come up with a different solution.

We could consider a `ssl.transport.mode = legacy` setting (default for now), that if used is the same as `off` but implies that `HBA` restrictions won't be used. 

Or we add a breaking changes entry to the release notes - so that users need to add a transport/trust entry if they have HBA enabled.

**Comment by BaurzhanSakhariev** [comment]
I see, thanks. I picked up legacy mode approach as it would be easier for now - we can make breaking change once we support optional SSL

File: docs/config/node.rst [hunk 1]
```diff
@@ -556,13 +556,15 @@ Layer Security (TLS).
...
-  | *Default:* ``off``
+  | *Default:* ``legacy``
   | *Runtime:* ``no``
 
   For communication between nodes, choose:
 
   ``off``
     SSL cannot be used
+  ``legacy``
+    SSL cannot be used and HBA will not be used even if HBA is enabled
```
Comment (mfussenegger) [inline docs/config/node.rst:567]:
```suggestion
    SSL is not used. If HBA is enabled, transport connections won't be verified. Any reachable host can establish a connection.
```

To clarify that this doesn't disable HBA entirely, but that it only affects how transport connections are treated.

**Review by mfussenegger — APPROVED** [review]
Left a suggestion, otherwise lgtm

**Pull request merged**